airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/base/AirbyteTracedException.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.base;
+
+import io.airbyte.protocol.models.AirbyteErrorTraceMessage;
+import io.airbyte.protocol.models.AirbyteErrorTraceMessage.FailureType;
+import io.airbyte.protocol.models.AirbyteMessage;
+import io.airbyte.protocol.models.AirbyteMessage.Type;
+import io.airbyte.protocol.models.AirbyteTraceMessage;
+import java.util.Arrays;
+import java.util.function.Consumer;
+
+public class AirbyteTracedException extends Exception {
+
+  private final Throwable e;
+  private final String displayMessage;
+  private final FailureType failureType;
+
+  public AirbyteTracedException(final Throwable e, final String displayMessage, final FailureType failureType) {
+    this.e = e;
+    this.displayMessage = displayMessage;
+    this.failureType = failureType;
+
+    // Not sure why defaultOutputRecordCollector is under Destination specifically,
+    // but this matches IntegrationRunner usage
+    Consumer<AirbyteMessage> outputRecordCollector = Destination::defaultOutputRecordCollector;
```

Comment:
This might be a separate issue, but from what I can tell, `defaultOutputrecordCollector` just uses `System.out.println` to print the json-formatted message without doing any sort of secret filtering, which was being done in the python CDK for log messages and now trace messages. Do we need to do anything like that here to prevent leaking secrets on java connectors as well, or is this handled elsewhere?